dolphin/drivers/huawei/oceanstor/oceanstor.py

```diff
@@ -0,0 +1,181 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from oslo_log import log
+from dolphin.common import constants
+from dolphin.drivers.huawei.oceanstor import rest_client, consts
+from dolphin.drivers import driver
+from dolphin import exception
+
+LOG = log.getLogger(__name__)
+
+
+class OceanStorDriver(driver.StorageDriver):
+    """OceanStorDriver implement Huawei OceanStor driver,
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.client = rest_client.RestClient(**kwargs)
+        self.client.login()
+        self.sector_size = consts.SECTORS_SIZE
+
+    def get_storage(self, context):
+
+        storage = self.client.get_storage()
+
+        # Get status
+        status = constants.StorageStatus.OFFLINE
+        if storage['RUNNINGSTATUS'] == consts.STATUS_STORAGE_NORMAL:
+            status = constants.StorageStatus.NORMAL
+
+        # Keep sector_size for use in list pools
+        self.sector_size = int(storage['SECTORSIZE'])
+
+        total_cap = int(storage['TOTALCAPACITY']) * self.sector_size
```

Comment:
Does it multiplies to actual capcity in bytes? sector_size is 512 .. what is the unit returned by rest client? Ifit is correct plz provide some comment here to show how the conversion is done..

Reply:
The rest client returns, size in 'sectors' as specified in OceanStor Interface reference doc. We need to multiply with sector size to get size in bytes.